packages/bpk-component-carousel/src/BpkCarousel.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useRef, useState } from 'react';
+
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import BpkPageIndicator, { VARIANT } from '../../bpk-component-page-indicator';
+import { cssModules } from '../../bpk-react-utils';
+
+import BpkCarouselContainer from './BpkCarouselContainer';
+import { useScrollToInitialImage } from './utils';
+
+import type { Props } from './types';
+
+import STYLES from './BpkCarousel.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+const BpkCarousel = ({
+  images,
+  indicatorClassName,
+  indicatorLabel = 'go to slide',
+  initialImageIndex = 0,
+  onImageChanged = null,
+}: Props) => {
+  const [shownImageIndex, updateShownImageIndex] = useState(initialImageIndex);
+  const imagesRef = useRef<Array<HTMLElement | null>>([]);
+
+  useScrollToInitialImage(initialImageIndex!, imagesRef);
+
+  return (
+    <>
+      <BpkCarouselContainer
+        images={images}
+        onVisible={updateShownImageIndex}
+        imagesRef={imagesRef}
+        onImageChanged={onImageChanged}
+      />
+      <BpkPageIndicator
+        currentIndex={shownImageIndex}
+        totalIndicators={images.length}
+        // eslint-disable-next-line @skyscanner/rules/forbid-component-props
+        className={getClassName('bpk-carousel-slides-indicator', indicatorClassName)}
```

Comment:
using `getClassName('bpk-carousel-slides-indicator'` to change the location of the page indicator on image, as `variant` in `BpkPageIndicator` is used to set the `background-color` of the dot, if removing it, we can't see it in Carousel, like
<img width="349" alt="image" src="https://github.com/Skyscanner/backpack/assets/139192353/1e63c65b-12af-4230-aaab-e0638f8b3c58">

Adding one more `indicatorClassName` is used to adjust the bottom's value, because bottom's value in `bpk-carousel-slides-indicator` does not meet the hotel funnel style.

Reply:
Thanks! This could indicate a potential issue then with the indicator component as this should have been built to work over images already, so we should probably look at this as part of this work to make sure it works as expected, as eventually className props will be removed from components to avoid CSS non-determinism issues

So we will need to fix the component to make this work :)